xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp

```diff
@@ -385,6 +385,9 @@ CDVDVideoCodec::VCReturn CDVDVideoCodecStarfish::GetPicture(VideoPicture* pVideo
   if (m_state == StarfishState::ERROR)
     return VC_ERROR;
 
+  if (m_state == StarfishState::ERROR)
```

Comment:
I don't think this condition could be true (or you would have returned two line up).
Maybe `StarfishState::EOS` would make more sense?